src/prefect/agent/docker/agent.py

```diff
@@ -403,6 +403,9 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
         container_mount_paths = self.container_mount_paths
         if container_mount_paths:
             host_config.update(binds=self.host_spec)
+        if run_config is not None and run_config.host_config is not None:
+            # The host_config passed from the run_config will overwrite defaults
+            host_config.update(run_config.host_config)
```

Comment:
No need for an explicit comparison to `None` (which is generally frowned upon in PEP recommendations, although you'll see mixed styles in this repo)
```suggestion
        if run_config is not None and run_config.host_config:
            # The host_config passed from the run_config will overwrite defaults
            host_config.update(run_config.host_config)
```

You could also do this, but I think the above is better
```python
        if run_config is not None:
            # The host_config passed from the run_config will overwrite defaults
            host_config.update(run_config.host_config or {})
```

Reply:
Would it be preferred to remove both of the comparisons to None? I was originally just trying to replicate how it worked with the env_vars. But I could also do something like this:
```suggestion
        if run_config and run_config.host_config:
            # The host_config passed from the run_config will overwrite defaults
            host_config.update(run_config.host_config)
```

Reply:
I will also take a closer look at the docker agent tests and see if there is a good place to check this. There wasn't an obvious one at a quick glance. Would you like me to write a new one if there aren't any good places to check the merged dictionary worked?

Reply:
The opinions here have varied on this one, hence the different messages :) I'm fine with either choice for the `host_config` value check, I'd rather not touch the `run_config` value check.

I think you can test this in `tests/agent/test_docker_agent.py::test_docker_agent_deploy_flow_run_config` or a copy of it extended for this case. `api.create_container.call_args[1]["host_config"]` should contain the settings of interest.

Reply:
@madkinsz So with the testing I am in a tricky spot. The api.create_container is a mock that returns a set value. Seems like the Docker SDK has a lot of overhead that isn't wanted in the tests. Which means it doesn't actually check if the dictionary merged into the host_config. The host config is also not returned anywhere else so also not possible to test. 

One option is to pull out the ~10 lines of code in the `deploy_flow` method of `DockerAgent` and create a new method called `create_host_config` that can be tested. What are your thoughts on that? These lines would get pulled out into their own function:

```
host_config = {
            "auto_remove": True,
            # Compatibility for linux -- https://github.com/docker/cli/issues/2290
            "extra_hosts": {"host.docker.internal": "host-gateway"},
        }  # type: dict
container_mount_paths = self.container_mount_paths
if container_mount_paths:
  host_config.update(binds=self.host_spec)
if run_config is not None and run_config.host_config:
  # The host_config passed from the run_config will overwrite defaults
  host_config.update(run_config.host_config)```

Reply:
I'm not opposed to moving it out if it's not feasible to check the mock, but it seems like it should be. . I see now that `api.create_container.call_args[1]["host_config"]` will have a mocked, fixed value but `api.create_host_config.call_args` should have the merged host config for inspection still.